lib/beaker/cli.rb

```diff
@@ -240,12 +240,58 @@ def preserve_hosts_file
       hosts_yaml['CONFIG'] = hosts_yaml['CONFIG'].merge(@options.reject{ |k,v| k =~ dontpreserve })
       # remove copy of HOSTS information
       hosts_yaml['CONFIG']['provision'] = false
+      hosts_yaml['HOSTS'].each do |ip,host|
+        if ( ! host.key?(:instance) )
+          next
+        end
+        ec2_instance = host[:instance]
+        new_instance = ec2_instance_to_hash(ec2_instance)
+        host[:instance] = new_instance
+      end
+
       File.open(preserved_hosts_filename, 'w') do |file|
         YAML.dump(hosts_yaml, file)
       end
       @options[:hosts_preserved_yaml_file] = preserved_hosts_filename
     end
 
+    # ec2_instance_to_hash
+    #
+    # converts an ec2 instance to a hash
+    # recurses into the instance variables looking for other AWS object to convert as well
+    # skips the configuration object because the AWS object doesn't write that to YAML
+    # This is specifically tailored to be a medium term solution to BKR-487
+    # It would be the wrong thing to use for anything else.
+    #
+    # @param obj
+    # @return [hash]
+    def ec2_instance_to_hash obj
+      h = Hash.new
+      if ( obj.class == AWS::Core::Configuration )
+        return obj
+      end
```

Comment:
should this conditional be after the one below as it's much less likely?

Reply:
no, because this condition is intended to stop execution from going into the conditional below.